client/verify.go

```diff
@@ -89,16 +89,21 @@ type resultWithPreviousSignature interface {
 	PreviousSignature() []byte
 }
 
-func asRandomData(r Result) *RandomData {
+func (v *verifyingClient) asRandomData(r Result) *RandomData {
 	rd, ok := r.(*RandomData)
 	if ok {
 		return rd
```

Comment:
With this line - don't we need now to give to all clients (grpc, http etc) the notion of this v2 signature and when to switch to it or not ? Otherwise this random data might contain v1 or v2 we don't know. That's what I wanted to avoid and only get this "selection logic" up the stack in one place (here). Or am i reading this wrong ?